code/infrastructure-api/src/kubeWatcher/kubeWatcher.js

```diff
@@ -7,20 +7,30 @@ import { parsePodLabels } from './kubernetesHelpers';
 import { CREATING, READY } from '../models/stack.model';
 import { STACKS, SELECTOR_LABEL } from '../stacks/Stacks';
 
-const kubeApi = config.get('kubernetesApi');
 const kubeNamespace = config.get('podNamespace');
-const events = ['added', 'modified', 'deleted'];
 const stackNames = Object.values(STACKS).map(stack => stack.name);
+const kc = new k8s.KubeConfig();
 
 function kubeWatcher() {
   logger.info(`Starting kube-watcher, listening for pods labelled "${SELECTOR_LABEL}" on "${kubeNamespace}" namespace.`);
 
-  return new KubeWatch('pods', {
-    url: kubeApi,
-    namespace: kubeNamespace,
-    labelSelector: SELECTOR_LABEL,
-    events,
-  });
+  kc.loadFromDefault();
+  const watch = new k8s.Watch(kc);
+  return watch.watch(`/api/v1/namespaces/${kubeNamespace}/pods`, { labelSelector: SELECTOR_LABEL },
+    (type, obj) => {
```

Comment:
Could you extract this block to a first class function please. For me that improves readability as the parameters to the `watch` function will be clearer.

I'm thinking
```
const url = '...',
const selector = { labelSelector: SELECTOR_LABEL };
watch.watch(url, selector, eventHandler, errorHandler);

function eventHandler(type, obj) { ... }
function errorHandler(err) { ... }
```

Reply:
Agreed. I'll make the changes.